### PR TITLE
fix(generate): fix style and template url references

### LINKED ISSUE #340: fix(generate/component): generated templateUrl and styleUrls paths are broken
Starting off with a fresh new A2 app using

```
$ ng new my-app
```

And creating a component using
```
$ ng g component test
```

Scaffolds the component fine in `src/app/test` however, `templateUrl` and `styleUrls` look like this:

```
templateUrl: 'app///test/test.html',
styleUrls: ['app///test/test.css']
```

angular-cli version: 0.0.26
node version: 4.3.2

### PR DESCRIPTION
**Pull request by Brocco**
Dynamic paths for components were incorrectly generating paths for templates and styles
Fixes #340

### CODE REVIEW

File: addon/ng2/blueprints/component/index.js [hunk 1]
```diff
@@ -13,7 +14,9 @@ module.exports = {
 
   locals: function(options) {
     return {
-      dynamicPath: this.dynamicPath.dir
+      dynamicPath: this.dynamicPath.dir === path.sep
```
Comment (hansl) [inline addon/ng2/blueprints/component/index.js]: this sounds like a `path.resolve(this.dynamicPath.dir)` or something like that, no?

**Comment by hansl** [comment]
Please add a test for this.

File: addon/ng2/blueprints/component/index.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 var stringUtils = require('ember-cli/lib/utilities/string');
 var dynamicPathParser = require('../../utilities/dynamic-path-parser');
+var path = require('path');
```
Comment (hansl) [inline addon/ng2/blueprints/component/index.js]: Nit: not needed anymore.

File: addon/ng2/utilities/dynamic-path-parser.js [hunk 3]
```diff
@@ -29,5 +29,9 @@ module.exports = function dynamicPathParser(project, entityName) {
 
   var parsedPath = path.parse(adjustedPath);
   
+  parsedPath.dir = parsedPath.dir === path.sep
```
Comment (hansl) [inline addon/ng2/utilities/dynamic-path-parser.js]: Nit: this all fit on one line.

**Comment by hansl** [comment]
2 nits, but LGTM. Stamping this. Good work! :)

**Comment by Brocco** [comment]
Thanks @hansl nits have been handled/updated

**Pull request merged**